src/vm/assembly.cpp

```diff
@@ -351,6 +364,10 @@ void Assembly::Terminate( BOOL signalProfiler )
         m_pClassLoader = NULL;
     }
 
+#ifdef FEATURE_EVENT_TRACE
+    g_cAssemblies--;
```

Comment:
Are you sure you need to subtract in both places? It looks like that it is subtracting twice to me.